bundles/org.openhab.core.addon.marketplace.karaf/src/main/java/org/openhab/core/addon/marketplace/karaf/internal/community/CommunityKarafAddonHandler.java

```diff
@@ -0,0 +1,189 @@
+/**
+ * Copyright (c) 2010-2021 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.core.addon.marketplace.karaf.internal.community;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.MalformedURLException;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.net.URL;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.StandardCopyOption;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Objects;
+import java.util.stream.Collectors;
+
+import org.apache.karaf.features.FeaturesService;
+import org.apache.karaf.features.Repository;
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.openhab.core.OpenHAB;
+import org.openhab.core.addon.Addon;
+import org.openhab.core.addon.marketplace.MarketplaceAddonHandler;
+import org.openhab.core.addon.marketplace.MarketplaceHandlerException;
+import org.osgi.service.component.annotations.Activate;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * A {@link CommunityKarafAddonHandler} implementation, which handles add-ons as kar files and installs them through
+ * the standard OSGi feature repository installation mechanism.
+ *
+ * @author Kai Kreuzer - Initial contribution and API
+ * @author Yannick Schaus - refactoring
+ * @author Jan N. Klug - refactor to support kar files
+ *
+ */
+@Component(immediate = true)
+@NonNullByDefault
+public class CommunityKarafAddonHandler implements MarketplaceAddonHandler {
+    private static final String KAR_CONTENT_TYPE = "application/vnd.openhab.feature;type=karfile";
+    private static final String BUNDLE_CACHE_PATH = OpenHAB.getUserDataFolder() + File.separator + "marketplace"
```

Comment:
Maybe it is possible to replace these strings with Java NIO `Path`? It's nicer to construct them with `Path.of(..)` and you would not have to convert between `File` and `Path` when using `Files`.

Reply:
Done.

Reply:
It certainly looks a lot nicer now. :slightly_smiling_face: :+1: